src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,220 @@
+                Optional<Car> optionalCar = get(resultSet.getObject("id", Long.class));
+                optionalCar.ifPresent(carList::add);
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get all data from DB", exception);
+        }
+        return carList;
     }
 
     @Override
     public Car update(Car car) {
-        List<Car> dbCar = Storage.getCars();
-        IntStream.range(0, dbCar.size())
-                .filter(i -> dbCar.get(i).equals(car))
-                .findFirst()
-                .ifPresent(i -> dbCar.set(i, car));
+        String queryForUpdateInCars = "update cars set model = ? where id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStmtCars = connection
+                            .prepareStatement(queryForUpdateInCars)) {
+            updateStmtCars.setString(1, car.getModel());
+            updateStmtCars.setLong(2, car.getId());
+
+            deleteDataFromCarsDrivers(connection, car);
+            updateStmtCars.executeUpdate();
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't update the data - "
+                    + car + "in table of cars or cars_drivers", exception);
+        }
+        createCarsDriversDB(car);
         return car;
     }
 
     @Override
     public boolean delete(Long id) {
-        return Storage.getCars().removeIf(i -> i.getId().equals(id));
+        String queryForCars = "UPDATE cars SET deleted = true where id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statementForCars = connection.prepareStatement(queryForCars)) {
+            statementForCars.setLong(1, id);
+            return statementForCars.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: " + id, exception);
+        }
     }
 
     public List<Car> getAllByDriver(Long driverId) {
-        return getAll().stream()
-                .filter(car -> car.getDrivers()
-                        .stream()
-                        .anyMatch(driver -> driver.getId().equals(driverId)))
-                .collect(Collectors.toList());
+        List<Car> carList = new ArrayList<>();
+        String query = "select cars.id as c_id, cars.model "
+                + " as c_m, drivers.id as d_id, drivers.name d_n,"
+                + " drivers.license_number as d_ln, manufacturers.id "
+                + " as m_id, manufacturers.name as m_n,"
+                + " manufacturers.country as m_c from cars"
+                + " inner join cars_drivers on cars.id = cars_drivers.car_id"
+                + " inner join drivers on drivers.id = cars_drivers.driver_id"
+                + " inner join manufacturers on manufacturers.id = cars.manufacturer_id"
+                + " where drivers.id = ?";
```

Comment:
same comment as for get() and getAll() methods - use 2 queries

in the first query you should use 3 joins (like here) and check that the car is not deleted and the driver is not deleted